atlascope/core/job_types/utils.py

```diff
@@ -10,3 +14,37 @@ def to_saveable_image(pil_image):
         output_bytes, None, 'file.png', 'image/png', output_bytes.getbuffer().nbytes, None
     )
     return output_file
+
+
+def save_output_dataset(original_dataset, job_name, output_image, outputs_dict):
+    outputs_dict['origin'] = f'Job Spawned at {timezone.now()}'
+    new_dataset = Dataset(
+        name=f'{original_dataset.name} {job_name}',
+        description=f'{job_name} for {original_dataset.name} as of {timezone.now()}',
+        public=original_dataset.public,
+        metadata=outputs_dict,
+        dataset_type='analytics',
+        source_dataset=original_dataset,
+    )
+    new_dataset.content.save(
+        f'{job_name.replace(" ", "_")}.png',
+        to_saveable_image(output_image),
+    )
+    new_dataset.save()
+
+    new_pin_location = [5, 5]
```

Comment:
I'm assuming that this pin location is relative to its `parent_dataset`? I just wanted to clarify. 

Also, any reason to have it start at (5, 5) and not (0, 0)? 

Having multiple pins in the same location might not ultimately be an issue, but I'm fine with this approach for now.